src/c++/library/common.h

```diff
@@ -330,10 +341,17 @@ class InferInput {
   /// \return Error object indicating success or failure.
   Error ByteSize(size_t* byte_size) const;
 
+  /// \return Error object indicating success or failure.
+  Error SetOptions(const std::shared_ptr<Options> options);
```

Comment:
I think it might be better to just copy the options instead of shared pointer.

Reply:
Irrelevant now 👍